### DISCUSSION TITLE
Pilot registration using QR codes

### DISCUSSION THREAD

**Discussion by chibaron** [RotorHazard/RotorHazard#505] [Ideas]
HI,
I'm testing a new feature I'd like to add.
What do you think?

This is a feature that automatically registers the pilot's name to the Betaflight splash image using a QR code.
Except for Betaflight, the QR code displayed on a smartphone or other device is captured by the FPV camera.

The modified FPV Receiver is connected to the raspberrypi for use
https://bit.ly/3qc15FN

Probably not going to use it in the race.
The best use of this feature is in practice sessions.
In practice, you'll be flying freely, without heat management.
I thought it would be great if the pilot name and time could be automatically aggregate.

https://youtu.be/9HhcGgAa2PY

![image](https://user-images.githubusercontent.com/15257530/107609350-03c93b80-6c82-11eb-926e-966426fcd8e2.png)

**Comment by HazardCreative** [comment]
Am I interpreting this right?
- Pilots replace their BetaFlight startup logo image with a QR code.
- A VRx scans frequencies looking for a quad that has just powered up.
- A processor detects the QR code and reads the pilot ID
- RH places the pilot with into the next heat using the detected channel

If so, that's a really interesting idea. I can see several drawbacks in practice which need overcome:
- Generating the QR codes
- Getting pilots to install them
- Getting the hardware and reader set up (looks like you have a good start but I'm wondering about other groups)
- Matching QR data with RH data
- Creating new heats on the fly (perhaps resolved by #264)

While I don't see this becoming core functionality, we're heading toward a global plugin system that would allow you to develop a module that anyone can download and add in if they desire. This seems like a perfect candidate for that.

**Reply by chibaron** [reply]
Thanks for the reply.
Your understanding is generally correct.

I also do competitive RC cars.
MyLaps is set up on a permanent RC car circuit.
Users can drive around the circuit at their leisure and see the results.
The results are automatically counted.
https://www.mylaps.com/jp/rc-drone-racing/

I'm wondering if RotorHazard can do something similar.
We need to bundle the results with the pilot's name, and this mechanism is the QR code.

To do this, RotorHazard needs to include a practice mode.
This is not a mode where several people start at the same time (heats).
It is more like a session where each person starts individually.
It is similar to livetime's practice mode.
Can you consider implementing this mode?

**Reply by HazardCreative** [reply]
Progress toward #264 is being made, but that's still heat-based check-in. Heat-based course utilization during practice is common in FPV because powering up when someone else is already flying is greatly frowned upon and batteries only last a few minutes.

**Reply by chibaron** [reply]
Thanks for the review.
I'll wait for progress on #264.

QR codes can be created with the link
https://chibaron.github.io/tinyviewpluspractice/tvp_qrcode.html

If this system is used by everyone, I'm thinking of including it as an OSD feature in betaflight.
